File: spring/actuator/src/main/java/com/hyfly/actuator/entity/validate/Diagnostic.java
package com.hyfly.actuator.entity.validate;

import com.alibaba.fastjson2.annotation.JSONField;
import lombok.Data;

@Data
public class Diagnostic {

    @JSONField(name = "severity")
    private String severity;

    @JSONField(name = "summary")
    private String summary;

    @JSONField(name = "detail")
    private String detail;

    @JSONField(name = "range")
    private Range range;

    @JSONField(name = "snippet")
    private Snippet snippet;
}

File: terraform/src/main/kotlin/com/hyfly/tf/actuator/entity/validate/Diagnostic.kt
package com.hyfly.tf.actuator.entity.validate

import com.alibaba.fastjson2.annotation.JSONField

class Diagnostic {
    @JSONField(name = "severity")
    val severity: String? = null

    @JSONField(name = "summary")
    val summary: String? = null

    @JSONField(name = "detail")
    val detail: String? = null

    @JSONField(name = "range")
    val range: Range? = null

    @JSONField(name = "snippet")
    val snippet: Snippet? = null
}
